apps/condo/domains/ticket/utils/helpers.ts

```diff
@@ -576,26 +578,43 @@ export const getChartOptions: IGetChartOptions = ({
     return { option, opts }
 }
 
+function getDeadlineStopPoint (ticket: ITicketUIState) {
+    const ticketStatusType = get(ticket, ['status', 'type'])
+    const ticketStatusUpdatedAt = get(ticket, ['statusUpdatedAt'])
+    let deadlineStopPoint = dayjs().startOf('day')
+
+    if (ticketStatusType === CLOSED_STATUS_TYPE || ticketStatusType === CANCELED_STATUS_TYPE) {
+        deadlineStopPoint = dayjs(ticketStatusUpdatedAt)
+    }
+
+    return deadlineStopPoint
+}
+
 export enum TicketDeadlineType {
     MORE_THAN_DAY,
     LESS_THAN_DAY,
     OVERDUE,
 }
 
-export function getDeadlineType (deadline: Dayjs): TicketDeadlineType {
-    const today = dayjs().startOf('day')
-    const tomorrow = today.add(1, 'day')
-    const isLessThanOneDay = deadline.isBefore(tomorrow) || deadline.isSame(tomorrow)
+export function getDeadlineType (ticket: ITicketUIState): TicketDeadlineType {
+    const deadline = dayjs(get(ticket, 'deadline'))
+    const deadlineStopPoint = getDeadlineStopPoint(ticket)
+
+    const deadlineStopPointNextDay = deadlineStopPoint.add(1, 'day')
```

Comment:
If more than 1 day is left before the deadline, then return the type `MORE_THAN_DAY`
Otherwise, the types are either `OVERDUE` or `LESS_THAN_DAY`.

Different messages are displayed depending on the return type

Reply:
Added doc to function